lib/VMwareWebService/broker_timeout.rb

```diff
@@ -0,0 +1,64 @@
+module Timeout
```

Comment:
Quite true. I now agree with @matthewd 's previous statement. The yield/lock window doesn't matter, given the client should have wrapped the call to timeout appropriately. And, the raise/kill window shouldn't matter, and only does because of the bug. So, should the `TODO`  comment in the code be more specific in describing the bug in question? It may be helpful in determining if this code is still needed for subsequent versions of Ruby.